### ISSUE TITLE
Add Updater + News in teeworlds

### ISSUE THREAD

**Issue by trafilaw** [teeworlds/teeworlds#2415]
Hey,

it would be awesome if there a Updater & News in teeworlds.
There are many players who still play with 0.7.3 and they are outdated.

Updater & News are nice in ddnet, so why not in 0.7 :D?

~Traffy

**Comment by Fisico** [comment]
Some kind of a news container would be cool. I'm not sure if an updater is needed, a lot of people play over steam and steam updates automatically.

**Comment by trafilaw** [comment]
Yea, but I saw today many players who played without steam and they didn't knew that teeworlds got a new update.

**Comment by Dune-jr** [comment]
Seems like many people play on 0.7.2 even.

**Comment by heinrich5991** [comment]
If the version server starts working again, they'll see a notice in the main menu.

**Comment by ChillerDragon** [comment]
> Seems like many people play on 0.7.2 even.

Seems like many people play on 0.6 even :trollface:

**Comment by Robyt3** [comment]
There is already #1199 and `PAGE_NEWS=0` has been sitting there for a while. Concerning that, teeworlds.com could just provide it's news a json resource, then we'd just have to add some HTTP implementation/dependency.

It would be interesting if all the data assets could be updated without manual releases, but I'm skeptical about automatically updating the executable.

**Comment by Dune-jr** [comment]
Yeah https://github.com/teeworlds/teeworlds/issues/1199 and https://github.com/teeworlds/teeworlds/issues/2380 would be enough imo

**Comment by fokkonaut** [comment]
@Robyt3 of course it works, ddnet also has it.

**Comment by Robyt3** [comment]
I'm not questioning that, I'm skeptical because of

- security implications
- people not actually wanting an executable to hotpatch itself 

Automatic updates (at least of the executable) are something that should be done through steam and other package manager IMO.

**Comment by Dune-jr** [comment]
That's a good point. I don't know any games that patch themselves, instead of being patched by a launcher, or requiring a redownload.

**Comment by trafilaw** [comment]
I'm not talking about a auto updater.
I'm talking about a button where you can update your client ;)

**Comment by Dune-jr** [comment]
@trafilaw what would this button do, exactly?

**Comment by Robyt3** [comment]
@Dune-jr https://github.com/ddnet/ddnet/blob/master/src/engine/client/updater.cpp

https://update4.ddnet.tw/update.json

Basically download a json manifest which defines file changes. Download client/server executable as tmp files, do some OS specific shenanigans to hotswap/replace the file while it's running and restart client with some more shenanigans.

**Comment by Learath2** [comment]
No shenanigans needed on most OSs, it's just that Windows XP doesn't seem to be reliable with replacing running executables.

**Comment by Dune-jr** [comment]
0.7 already broke Windows XP support afaik
